src/ferc_xbrl_extractor/instance.py

```diff
@@ -230,13 +230,15 @@ def __init__(
             filing_name: Name of parsed filing.
         """
         # This is a nested dictionary of dictionaries to locate facts by context
-        self.instant_facts = {}
-        self.duration_facts = {}
+        self.instant_facts: dict[tuple, dict[Context, list[Fact]]] = {}
+        self.duration_facts: dict[tuple, dict[Context, list[Fact]]] = {}
```

Comment:
What a type hint! What are the keys of these dicts?

Reply:
Whew yeah, that looks like it might be a candidate for a typed dict or Pydantic class or something.

Reply:
I've added a type alias with some explanation of how it's used. I'm not sure that a pydantic class or typed dict would be useful because the keys are pretty dynamic. Let me know if this clarifies things at all